Tella/Data/Networking/Repositories/GDriveRepository.swift

```diff
@@ -0,0 +1,450 @@
+                    else {
+                        return
+                    }
+
+                    let sharedDrives = drives.map { drive in
+                        SharedDrive(
+                            id: drive.identifier ?? "", name: drive.name ?? "",
+                            kind: drive.kind ?? "")
+                    }
+
+                    promise(.success(sharedDrives))
+                }
+            }
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    func createDriveFolder(
+        folderName: String,
+        parentId: String? = nil,
+        description: String?
+    ) -> AnyPublisher<String, APIError> {
+        Deferred {
+            Future { promise in
+                guard self.networkMonitor.isConnected else {
+                    promise(.failure(.noInternetConnection))
+                    return
+                }
+                Task {
+                    do {
+                        try await self.ensureSignedIn()
+                        self.performCreateDriveFolder(
+                            folderName: folderName,
+                            parentId: parentId,
+                            description: description,
+                            promise: promise)
+                    } catch {
+                        promise(.failure(self.mapToAPIError(error)))
+                    }
+                }
+            }
+        }.eraseToAnyPublisher()
+    }
+    
+    private func performCreateDriveFolder(
+        folderName: String,
+        parentId: String?,
+        description: String?,
+        promise: @escaping (Result<String, APIError>) -> Void
+    ) {
+        guard let user = googleUser else {
+            promise(.failure(APIError.noToken))
+            return
+        }
+
+        let driveService = GTLRDriveService()
+        driveService.authorizer = user.fetcherAuthorizer
+
+        let folder = GTLRDrive_File()
+        folder.name = folderName
+        folder.mimeType = GoogleAuthConstants.gDriveFolderMimeType
+
+        if let parentID = parentId {
+            folder.parents = [parentID]
+        }
+        
+        if let description = description {
+            folder.descriptionProperty = description
+        }
+
+        let query = GTLRDriveQuery_FilesCreate.query(withObject: folder, uploadParameters: nil)
+        query.supportsAllDrives = true
+        
+        driveService.executeQuery(query) { (ticket, file, error) in
+            if let error = error {
+                print("Error creating folder: \(error.localizedDescription)")
+                promise(.failure(self.mapToAPIError(error)))
+                return
+            }
+
+            guard let createdFile = file as? GTLRDrive_File else {
+                promise(.failure(APIError.unexpectedResponse))
+                return
+            }
+
+            promise(.success(createdFile.identifier ?? ""))
+        }
+    }
+    
+    func uploadFile(
+            fileURL: URL,
```

Comment:
Instead to pass many parameters in the function what do you think if you pass a struct that contains all the parameters